extensions/ql-vscode/src/codeql-cli/cli.ts

```diff
@@ -1605,8 +1606,8 @@ export function spawnServer(
   command: string[],
   commandArgs: string[],
   logger: Logger,
-  stderrListener: (data: any) => void,
-  stdoutListener?: (data: any) => void,
+  stderrListener: (data: unknown) => void,
+  stdoutListener?: (data: unknown) => void,
```

Comment:
I think we can change this to more strict types since [the documentation mentions that this can only be a string or `Buffer`](https://nodejs.org/api/stream.html#event-data), unless the stream is operating in object mode. A child process's stdout or stderr will never operate in object mode, so we should be able to change this to:

```suggestion
  stderrListener: (data: string | Buffer) => void,
  stdoutListener?: (data: string | Buffer) => void,
```

I don't get any TypeScript errors locally when doing this, so it seems like narrowing the types here should work without giving any problems. I don't think we need the `asString` function in that case.

Reply:
Thanks for linking to the docs. That makes sense to me. It's a shame they don't have separate type information for non-object mode streams, and therefore all streams have to default to `any`.

I'll take a look at using `string | Buffer` and get rid of the `asString` method.